apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -0,0 +1,848 @@
+*/
+static void utc_audio_pcm_readi_p(void)
+{
+	int fd;
+	char input_str[16];
+	ssize_t size;
+	ssize_t bytes;
+	unsigned int frames_read = 0;
+	unsigned int total_frames_read = 0;
+	unsigned int rate;
+	int capturing;
+	int ret;
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	g_record_buffer = (char *)malloc(bytes); //rate * bit * channels * duration
+	TC_ASSERT("pcm_readi_p", g_record_buffer);
+
+	printf("Record will be start for 3s, press any key to start\n");
+	fflush(stdout);
+	if (gets(input_str) == NULL) {
+		TC_ASSERT("audio_pcm_readi", gets(input_str));
+		free(g_record_buffer);
+		return;
+	}
+
+	capturing = 1;
+	while (capturing) {
+		frames_read = pcm_readi(g_pcm, (void *) &g_record_buffer[total_frames_read], size);
+		total_frames_read += frames_read;
+		if ((total_frames_read / rate) >= AUDIO_RECORD_DURATION) {
+			capturing = 0;
+		}
+	}
+	pcm_close(g_pcm);
+	fd = open(AUDIO_TEST_FILE, O_RDWR | O_CREAT);
+	ret = write(fd, g_record_buffer, bytes);
+	TC_ASSERT_GT_CLEANUP("pcm_reaid_p", ret, 0, close(fd));
+	close(fd);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_readi_n
+* @brief            read captured data from chip
+* @scenario         call readi with invalid values
+* @apicovered       pcm_readi
+* @precondition     pcm should be opened before
+* @postcondition    NA
+*/
+static void utc_audio_pcm_readi_n(void)
+{
+	int ret;
+	ssize_t size;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_GT("pcm_readi", pcm_get_file_descriptor(g_pcm), 0);
+
+	size = pcm_get_buffer_size(g_pcm);
+
+	ret = pcm_readi(NULL, g_record_buffer, size);
+	TC_ASSERT_LT("pcm_readi", ret, 0);
+
+	ret = pcm_readi(g_pcm, NULL, size);
+	TC_ASSERT_LT("pcm_readi", ret, 0);
+
+	ret = pcm_readi(g_pcm, g_record_buffer, 0);
+	TC_ASSERT_LT("pcm_readi", ret, 0);
+
+	pcm_close(g_pcm);
+	free(g_record_buffer);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_write_p
+* @brief            play captured data from filesystem
+* @scenario         play recored file previously
+* @apicovered       pcm_writei
+* @precondition     pcm_readi should be opened before
+* @postcondition    NA
+*/
+static void utc_audio_pcm_writei_p(void)
+{
+	int fd;
+	ssize_t size;
+	ssize_t bytes;
+	int ret;
+	unsigned int total_frames_write = 0;
+	unsigned int rate;
+	int play;
+
+	fd = open(AUDIO_TEST_FILE, O_RDONLY);
+	TC_ASSERT_GEQ("pcm_writei", fd, 0);
+
+	/* use default config here */
+	g_pcm = pcm_open(0, 0, PCM_OUT, NULL);
+	TC_ASSERT_GT("pcm_writei", pcm_get_file_descriptor(g_pcm), 0);
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	g_record_buffer = (char *)malloc(bytes); //rate * bit * channels * duration
+	TC_ASSERT_CLEANUP("pcm_readi_p", g_record_buffer, close(fd));
```

Comment:
Cleanup fd and pcm. Same at below as this

Reply:
done